backend/backend/Controllers/VideosController.cs

```diff
@@ -12,16 +18,246 @@ namespace backend.Controllers
     public class VideosController : ControllerBase
     {
         private readonly BackendContext _db;
+        private readonly string _uploadPath;
+        private readonly string _tempPath;
+        private readonly int _chunkSize;
 
         public VideosController(BackendContext context)
         {
             _db = context;
+            _uploadPath = Path.Combine(Directory.GetCurrentDirectory(), "Uploads");
+            _tempPath = Path.Combine(_uploadPath, "Temp");
+            _chunkSize = 28000000; // 28MB
         }
 
         [HttpGet,Route("all")]
         public ActionResult<List<Video>> GetAllVideos()
         {
             return _db.Videos.ToList();
         }
+
+        [HttpPost, Route("UploadChunks")]
+        public async Task<IActionResult> UploadChunk(string chunkNumber, string fileName)
+        {
+            try
+            {
+                string newPath = Path.Combine(_tempPath, Path.GetFileNameWithoutExtension(fileName) + "-" + chunkNumber + Path.GetExtension(fileName));
```

Comment:
I'm gonna comment agressively, but probably because of time constraints most of this won't matter. What if two users are uploading a file that has the same name and extension?